cetlvast/CMakeLists.txt

```diff
@@ -0,0 +1,132 @@
+    set(CMAKE_MODULE_PATH ${CETLVAST_CMAKE_MODULE_PATH})
+else()
+    #
+    # Tell cmake where to find our custom scripts.
+    #
+    set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")
+    message(STATUS "CMAKE_MODULE_PATH set to ${CMAKE_MODULE_PATH}")
+endif()
+
+# +---------------------------------------------------------------------------+
+# | OPTIONS/VARIANTS
+# +---------------------------------------------------------------------------+
+
+option(CETL_ENABLE_DEBUG_ASSERT "Enable or disable runtime asserts in CETL code." OFF)
+option(CETLVAST_ENABLE_COVERAGE "Enable or disable coverage data instrumentation in test builds." OFF)
+option(CETLVAST_STYLE_CHECK "Enable or disable a check for violations of the project's style standards." ON)
+
+set(CETLVAST_CPP_STANDARD "20" CACHE STRING "C++ standard to use when compiling.")
+set_property(CACHE CETLVAST_CPP_STANDARD PROPERTY STRINGS 14 17 20 23)
+
+message(STATUS "[ Build Options ]--------------------------------------\n\
+    CMAKE_BUILD_TYPE:         ${CMAKE_BUILD_TYPE}\n\
+    CETL_ENABLE_DEBUG_ASSERT: ${CETL_ENABLE_DEBUG_ASSERT}\n\
+    CETLVAST_ENABLE_COVERAGE: ${CETLVAST_ENABLE_COVERAGE}\n\
+    CETLVAST_STYLE_CHECK:     ${CETLVAST_STYLE_CHECK}\n\
+    CETLVAST_CPP_STANDARD:    ${CETLVAST_CPP_STANDARD}\n\
+-----------------------------------------------------------\n\
+")
+
+# +---------------------------------------------------------------------------+
+# | KITS
+# +---------------------------------------------------------------------------+
+
+if(NOT DEFINED CETLVAST_FLAG_SET)
+    message(VERBOSE "CETLVAST_FLAG_SET was not defined. Setting to \"native-full\" as a default")
+    set(CETLVAST_FLAG_SET ${CMAKE_SOURCE_DIR}/cmake/compiler_flag_sets/native-full.cmake)
+else()
+    set(CETLVAST_FLAG_SET ${CETLVAST_FLAG_SET})
+endif()
+
+if(NOT DEFINED CETLVAST_TEST_SUITE)
+    message(VERBOSE "CETLVAST_TEST_SUITE was not defined. Setting to \"native tests\" as a default")
+    set(CETLVAST_TEST_SUITE ${CMAKE_SOURCE_DIR}/cmake/suites/native.cmake)
+else()
+    set(CETLVAST_TEST_SUITE ${CETLVAST_TEST_SUITE})
+endif()
+
+message(STATUS "[ Build Kit ]-------------------------------------------\n\
+    CMAKE_TOOLCHAIN_FILE:     ${CMAKE_TOOLCHAIN_FILE}\n\
+    CETLVAST_FLAG_SET:        ${CETLVAST_FLAG_SET}\n\
+    CETLVAST_TEST_SUITE:      ${CETLVAST_TEST_SUITE}\n\
+-----------------------------------------------------------\n\
+")
+
+# +---------------------------------------------------------------------------+
+# | FLAG SETS
+# +---------------------------------------------------------------------------+
+#
+# Load all our compile flag sets into the appropriate places.
+#
+include(${CETLVAST_FLAG_SET})
+
+# +---------------------------------------------------------------------------+
+# | EXTERNAL DEPENDENCIES
+# +---------------------------------------------------------------------------+
+
+#
+# googletest (and googlemock) external project.
+#
+find_package(gtest REQUIRED)
```

Comment:
eh, I should move this into the unittest suite.cmake. It's not used for any other test suite.